portfolio/src/main/java/com/google/sps/servlets/UserInfoServlet.java

```diff
@@ -36,28 +36,29 @@
 @WebServlet("/user-info")
 public class UserInfoServlet extends HttpServlet {
   private UserDaoDatastore userStorage = new UserDaoDatastore();
+  private Gson gson = new Gson();
 
   @Override
   public void init() {
     UserDaoDatastore userStorage = new UserDaoDatastore();
   }
 
-  private String toJson(String status) {
-    Gson gson = new Gson();
-    return gson.toJson(status);
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json");
     UserService userService = UserServiceFactory.getUserService();
     if (userService.isUserLoggedIn()) {
-      String nickname = getUserNickname(userService.getCurrentUser().getEmail());
-      response.getWriter().println(toJson(nickname));
+        String email = userService.getCurrentUser().getEmail(); // user email is used as id
+        Optional<User> user = userStorage.get(email);
+        if (user.isPresent()) {
+            response.getWriter().println(gson.toJson(user.get()));
+        } else {
+            response.getWriter().println(gson.toJson(user.empty()));
```

Comment:
How the code is structured is that user info isn't saved until users want them to be, which is when they set up a nickname. Because of this, userStorage.get(email) returns an Optional<User>, so the user could be  logged in, but not yet saved in datastore. user.isPresent() checks if the value in Optional exists, and if it doesn't(the user hasn't saved their info/set their personal info yet), then an empty Optional instance is returned, and on the page, Hello {{user nickname}} won't yet be shown.

Reply:
Cool, that all makes sense! 

But we do at least know their email, so I feel we should return them a user object (with only the email set) and we could show that instead of the nickname, until they set a nickname up!